### PR TITLE
Add column filtering on SCA all buses

### PR DESCRIPTION
**Pull request by flomillot**
Add the filtering on all buses SCA.
Code to be replaced by full custom headers.

### CODE REVIEW

File: src/components/utils/aggrid/custom-set-filter.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    forwardRef,
+    useEffect,
+    useImperativeHandle,
+    useState,
+} from 'react';
+import { IDoesFilterPassParams, IFilterParams } from 'ag-grid-community';
+import { Option } from 'components/results/shortcircuit/shortcircuit-analysis-result.type';
+
+// TODO temporary, remove it when SCA results will use full custom headers and filters
+export default forwardRef((props: IFilterParams, ref) => {
```
Comment (sBouzols) [inline src/components/utils/aggrid/custom-set-filter.tsx:18]: Give this component a default name ? CustomSetFilter ?

File: src/components/utils/aggrid/custom-set-filter.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    return (
+        <div
+            style={{
+                display: 'flex',
+                flexDirection: 'column',
+                padding: 8,
+            }}
+        >
+            <div style={{ fontWeight: 'bold', margin: 2 }}>Select values</div>
```
Comment (sBouzols) [inline src/components/utils/aggrid/custom-set-filter.tsx:77]: You should use MUI Box, Typography and theme spacing, but I know it's temporary...

File: src/components/results/shortcircuit/shortcircuit-analysis-result.tsx [hunk 3]
```diff
@@ -108,9 +128,9 @@ export const ShortCircuitAnalysisResult: FunctionComponent<
...
             selector,
         })
-            .then((result: SCAPagedResults) => {
+            .then((result: SCAPagedResults | null) => {
                 if (active) {
-                    const { content, totalElements } = result;
+                    const { content = [], totalElements } = result || {};
                     updateResult(content);
                     if (totalElements && content.length) {
                         setCount(totalElements);
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.tsx:136]: Honestly I kept the previous behaviour

File: src/components/results/shortcircuit/shortcircuit-analysis-result.tsx [hunk 4]
```diff
@@ -139,13 +159,54 @@ export const ShortCircuitAnalysisResult: FunctionComponent<
...
         analysisType,
+        analysisStatus,
         updateResult,
         studyUuid,
         currentNode?.id,
         intl,
         shortCircuitNotif,
     ]);
 
+    useEffect(() => {
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.tsx:170]: Sonar wants it.

File: src/components/utils/aggrid/custom-set-filter.tsx (hunk 1 above)
Comment (flomillot) [reply src/components/utils/aggrid/custom-set-filter.tsx:18]: Ok done

File: src/components/results/shortcircuit/shortcircuit-analysis-one-bus-result.tsx [hunk 5]
```diff
@@ -57,7 +65,13 @@ export const ShortCircuitAnalysisOneBusResult = () => {
             }
             setFaultResult(result.faults[0]);
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/results/shortcircuit/shortcircuit-analysis-one-bus-result.tsx:66]: Fixed

File: src/components/utils/aggrid/custom-set-filter.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    style={{ display: 'flex', alignItems: 'center', margin: 1 }}
+                >
+                    <input
+                        style={{ marginRight: 7 }}
+                        type="checkbox"
+                        name="filter"
+                        value={option.value}
+                        checked={selectedValues.includes(option.value)}
+                        onChange={onChange}
+                    />{' '}
```
[the remark this answers is not in the document]
Comment (flomillot) [reply src/components/utils/aggrid/custom-set-filter.tsx:90]: It's prettier who added it

File: src/components/results/shortcircuit/shortcircuit-analysis-result.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (sBouzols) [reply src/components/results/shortcircuit/shortcircuit-analysis-result.tsx:168]: My bad that's true

**Review by sBouzols — APPROVED** [review]
Code review OK
Tests OK
Console Warning Check OK

**Review by anistouri — APPROVED** [review]

**Pull request merged**